tests/standalone_plugins/test_deprecate_vts.py

```diff
@@ -0,0 +1,164 @@
+            testfile2.touch()
+
+            to_deprecate = [
+                DeprecatedFile(
+                    name="testfile1.nasl",
+                    full_path=testfile1,
+                    content=NASL_CONTENT,
+                )
+            ]
+            deprecate(out_dir, to_deprecate)
+
+            result = testfile2.read_text(encoding="utf8")
+            self.assertNotIn(result, "script_mandatory_keys")
+            self.assertNotIn(result, "script_dependencies")
+            self.assertNotIn(result, 'include("revisions-lib.inc");')
+            assert "This VT has been deprecated." in result
+
+    def test_deprecate_kb_item(self):
+        with TemporaryDirectory() as out_dir, TemporaryDirectory() as in_dir:
+            testfile1 = in_dir / "testfile1.nasl"
+            testfile1.write_text(NASL_CONTENT_KB, encoding="utf8")
+
+            to_deprecate = [
+                DeprecatedFile(
+                    name="testfile1.nasl",
+                    full_path=testfile1,
+                    content=NASL_CONTENT_KB,
+                )
+            ]
+            deprecate(out_dir, to_deprecate)
+            self.assertLogs(
+                "Unable to deprecate testfile1.nasl. There are still KB keys remaining."
+            )
+
+    def test_get_summary(self):
+        result = get_summary(NASL_CONTENT)
+        expected = (
+            "The remote host is missing an update for the 'gd'\n  package(s) announced "
+            "via the RHSA-2020:5443-01 advisory."
+        )
+        self.assertEqual(result, expected)
+
+    def test_finalize_content(self):
+        result = finalize_content(NASL_CONTENT)
+        expected = (
+            '...if(description)\n{\n  script_oid("1.3.6.1.4.1.25623.1.0.910673");\n  '
+            'script_version("2024-03-12T14:15:13+0000");\n  script_name("RedHat: Security Advisory for gd (RHSA-2020:5443-01)");\n  script_family("Red Hat Local Security Checks");\n  script_dependencies("gather-package-list.nasl");\n  script_mandatory_keys("ssh/login/rhel", "ssh/login/rpms", re:"ssh/login/release=RHENT_7");\n\n  script_xref(name:"RHSA", value:"2020:5443-01");\n  script_xref(name:"URL", value:"https://www.redhat.com/archives/rhsa-announce/2020-December/msg00044.html");\n\n  script_tag(name:"summary", value:"The remote host is missing an update for the \'gd\'\n  package(s) announced via the RHSA-2020:5443-01 advisory.");\n\n  script_tag(name: \'deprecated\', value: TRUE);\n\nexit(0);\n}\n\nexit(66);\n'
+        )
+        self.assertEqual(result, expected)
+
+    def test_update_summary_no_oid_match(self):
+        file = DeprecatedFile(
+            name="testfile.nasl",
+            full_path=Path("dir/testfile.nasl"),
+            content=NASL_CONTENT,
+        )
+        result = update_summary(file)
+        expected = (
+            "This VT has been deprecated.The remote host is missing an update for the 'gd'\n "
+            " package(s) announced via the RHSA-2020:5443-01 advisory."
```

Comment:
I would suggest to add the deprecation notice at the end of the summary tag and just use a more generic text like e.g.:

> The remote host is missing an update for the 'gd' package(s) announced via the RHSA-2020:5443-01 advisory.
>
> Note: This VT has been deprecated and replaced by a Notus scanner based one.

or similar.

Reply:
Personally i would be absolutely fine to just use a general more standardized format for all Notus relevant deprecated LSCs, a user should be clever enough to add e.g. the Advisory-ID into the SecInfo search to find the replacements.

Note that such "Notus scanner based one" text should only be added if that applies to the deprecated VT.

Reply:
Sounds great :+1: